### PR TITLE
Only display the plugin's settings errors, not all

### LINKED ISSUE #11: "Settings saved" notice is displayed twice
bildschirmfoto 2016-12-07 um 22 02 41

### PR DESCRIPTION
**Pull request by swissspidy**
Fixes #11.

The settings might be duplicated because another plugin of ours interferes with it and is calling just `settings_errors()` as well.

### CODE REVIEW

**Comment by ocean90** [comment]
This PR doesn't fix it for me.

Steps to reproduce:

* `$ wp option delete rplus_wp_widget_disable_sidebar_option`
* Go to wp-admin/themes.php?page=wp-widget-disable
* Select a widget
* Click "Save Changes"

**Comment by ocean90** [comment]
`sanitize_sidebar_widgets()` is called twice because `update_option()` calls `add_option()` if the option doesn't exist yet.

**Comment by neverything** [comment]
Was that the culprit all along?

**Comment by swissspidy** [comment]
No wonder I couldn't reproduce it when I never tried to remove that option 😅

Will fix that in this PR as soon as possible.

**Comment by ocean90** [comment]
@neverything Probably the other half of the issue. Only returning plugin's own errors makes kind of sense too. :) Related ticket: https://core.trac.wordpress.org/ticket/21989

I guess the only fix is to store the default values in the option when the plugin is activated.

File: classes/class-wp-widget-disable.php [hunk 1]
```diff
@@ -55,6 +55,19 @@ class WP_Widget_Disable {
...
+	 * Saves empty values for the plugin's options upon plugin activation.
+	 *
+	 * @link https://core.trac.wordpress.org/ticket/21989
+	 * @link https://github.com/wearerequired/WP-Widget-Disable/issues/11
+	 *
+	 * @since 1.6.1
+	 */
+	public function set_default_options() {
+		add_option( $this->sidebar_widgets_option, array() );
+		add_option( $this->dashboard_widgets_option, array() );
```
Comment (ocean90) [inline classes/class-wp-widget-disable.php]: Shouldn't this only happen if the options don't exist yet?

Comment (swissspidy) [inline classes/class-wp-widget-disable.php]: You're right. Fixed.

File: classes/class-wp-widget-disable.php [hunk 2]
```diff
@@ -55,6 +55,24 @@ class WP_Widget_Disable {
 	protected $page_hook = '';
 
 	/**
+	 * Saves empty values for the plugin's options upon plugin activation.
+	 *
+	 * @link https://core.trac.wordpress.org/ticket/21989
+	 * @link https://github.com/wearerequired/WP-Widget-Disable/issues/11
+	 *
+	 * @since 1.6.1
```
Comment (ocean90) [inline classes/class-wp-widget-disable.php]: Should be 1.7.0

**Pull request merged**